### PR TITLE
DOC: Fix various warnings

### PR DESCRIPTION
**Pull request by TomAugspurger**

### CODE REVIEW

File: doc/source/whatsnew/v0.10.0.txt [hunk 1]
```diff
@@ -409,7 +409,7 @@ N Dimensional Panels (Experimental)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Adding experimental support for Panel4D and factory functions to create n-dimensional named panels.
-:ref:`Docs <dsintro.panel4d>` for NDim. Here is a taste of what to expect.
+`Docs` for NDim. Here is a taste of what to expect.
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.10.0.txt]: I would just remove the "Docs for NDim" part

File: doc/source/whatsnew/v0.16.1.txt [hunk 2]
```diff
@@ -26,7 +26,7 @@ Highlights include:
 
 .. warning::
 
-   In pandas 0.17.0, the sub-package ``pandas.io.data`` will be removed in favor of a separately installable package. See :ref:`here for details <remote_data.pandas_datareader>` (:issue:`8961`)
+   In pandas 0.17.0, the sub-package ``pandas.io.data`` will be removed in favor of a separately installable package. See `here for details` (:issue:`8961`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.16.1.txt]: same here, I would just remove the "See here for details"

**Comment by TomAugspurger** [comment]
OK, if this all passes I think it's ready. There are still a few warnings left, but those are more involved.

**Comment by TomAugspurger** [comment]
I added a toctree for the docstring guide. Here's the rendered version

screen shot 2018-03-28 at 1 21 42 pm

**Pull request merged**